Add Model contracts section (#3305)
dbt-labs/docs.getdbt.com: website/docs/reference/resource-configs/trino-configs.md

```
</File>

## Model contracts

> The `dbt-trino` adapter supports [Model contracts](/docs/collaborate/govern/model-contracts). However, currently only `not_null` [constraint](/reference/resource-properties/constraints) is supported.

Reviewer: small sugg for a better clarity. 

```suggestion
The `dbt-trino` adapter supports [model contracts](/docs/collaborate/govern/model-contracts). Currently, only [constraints](/reference/resource-properties/constraints) with `type` as `not_null` are supported.
```

> The `dbt-trino` adapter supports [Model contracts](/docs/collaborate/govern/model-contracts). However, currently only `not_null` [constraint](/reference/resource-properties/constraints) is supported.

Reviewer 2: ```suggestion
[Model contracts](/docs/collaborate/govern/model-contracts) are supported. However the only supported constraint is `not_null` [contracts: constraints](/reference/resource-properties/constraints)
```

> Please ensure that underlying connector supports `not null`, before using this constraint in your model.

Reviewer: ```suggestion
Before using a constraint in your model, make sure the underlying connector supports `not_null`.
```

Reviewer: curious, what happens if it doesn't? what sort of issues will the user encounter if they do end up using it? it's good practice to inform people _why_ they should do a thing as it helps with learning.

Author: While trying `not_null` constraint on connector which doesn't support it, user will receive Trino error at dbt run stage, e.g.:
`Hive tables do not support NOT NULL columns`
Should I add detailed description in docs what would happen then? I think that it is not necessary in this case, as making sure that you use supported feature is self explanatory IMHO - you can't use feature, which is not supported 🙂 
WDYT?

> Before using `not_null` constraint in your model, make sure the underlying connector supports `not null`.

Reviewer: you're right, @damian3031 , that would be too much. how about this or something similar? 

```suggestion
Before using `not_null` constraints in your model, make sure the underlying connector supports `not null` to avoid running into errors.```

Author: adjusted, thanks!
